tests/tests_default.yml

```diff
@@ -1,8 +1,14 @@
 - name: Ensure that the role runs with default parameters
   hosts: all
   become: true
+  vars:
+    __rsyslog_version: "N/A"
 
   tasks:
     - name: default run (NOOP)
       include_role:
         name: linux-system-roles.logging
+
+    - name: Check rsyslog version
+      assert:
+        that: __rsyslog_version != "N/A" and __rsyslog_version != "0.0.0"
```

Comment:
Sorry for being late to the party, but does this assert belong here? For all other roles the default test just ensures that the role does not crash when no parameters are supplied. Is there a reason for the logging role to be different? Also, is it a good idea to test for an internal variable? I think it would be better to check the actual effect of the role on the system in tests rather than checking that the role has set some variable which the user will not use anyway - such a test is rather artificial.

Reply:
> Sorry for being late to the party, but does this assert belong here? For all other roles the default test just ensures that the role does not crash when no parameters are supplied. Is there a reason for the logging role to be different? Also, is it a good idea to test for an internal variable? I think it would be better to check the actual effect of the role on the system in tests rather than checking that the role has set some variable which the user will not use anyway - such a test is rather artificial.

Thanks for your comments, @pcahyna. They are valuable at any timing!
1) I will remove the check from tests_defaults.yml, which is not necessary.
2) You might remember that in the beginning, I was writing the version number at the top of rsyslog.conf anc checking the value in the test playbooks, which you recommended to replace with some other way. This is the second attempt. All I want to check is the version is set to the internal variable __rsyslog_version and the value is valid... If I have to, the third attempt might be using the `fail` module in the task (not in the test code) - if the value of __rsyslog_version looks invalid, we could fail there. Do we want to make the change?

Reply:
@nhosoi I don't understand the importance of checking, in the tests, that the value of `__rsyslog_version` is valid. The variable is not going to be used by the users, why is it important that it is valid?
If you want to make sure it is being set in a sensible way, you can add a separate unit test that checks for it (the current test are more integration tests, but adding a unit test is valid, too. I just don't want to mix those two test types).

Reply:
I think though that it would be good to review first the need for this variable - can't it be simply eliminated?

Reply:
> I think though that it would be good to review first the need for this variable - can't it be simply eliminated?

It is likely needed in this effort.
RHELPLAN-32352 - Add performance tuning parameters
Q. how do we handle rhel-7 and rhel-8 version difference?

Reply:
Opened a PR https://github.com/linux-system-roles/logging/pull/125.
Executing "Check rsyslog version" test just in tests_version.yml and eliminating it from the rest of the test cases.

Reply:
>  Q. how do we handle rhel-7 and rhel-8 version difference?

https://github.com/oasis-roles/meta_standards/pull/16/files#diff-04c6e90faac2675aa89e2176d2eec7d8R184

so you would have `vars/RedHat_7.yml` and `vars/RedHat_8.yml` where you would set platform/version specific variables.  You could do the same with tasks e.g. have `tasks/setup_RedHat_7.yml` etc.  If CentOS works exactly the same way, then you could just symlink `vars/CentOS_7.yml` to `vars/RedHat_7.yml` etc.

I suppose if there are differences between RHEL 7.8 and 7.9, or RHEL 8.2 and RHEL 8.3, we could use `ansible_distribution_minor_version` too

Reply:
If there are differences between minor versions, it may be in fact preferable to check the version of rsyslog rather than the distribution minor version. (What if the user updates just some of packages?) Is it really the case though? If not, it would be better to discuss the problem only when it really occurs.